plugins/modules/flatpak.py

```diff
@@ -195,6 +226,26 @@ def install_flat(module, binary, remote, names, method, no_dependencies):
     result['changed'] = True
 
 
+def update_flat(module, binary, names, method, no_dependencies):
+    """Update existing flatpaks."""
+    global result  # pylint: disable=global-variable-not-assigned
+    installed_flat_names = [
+        _match_installed_flat_name(module, binary, name, method)
+        for name in names
+    ]
+    command = [binary, "update", "--{0}".format(method)]
+    flatpak_version = _flatpak_version(module, binary)
+    if LooseVersion(flatpak_version) < LooseVersion('1.1.3'):
+        command += ["-y"]
+    else:
+        command += ["--noninteractive"]
+    if no_dependencies:
+        command += ["--no-deps"]
+    command += installed_flat_names
+    stdout = _flatpak_command(module, module.check_mode, command)
+    result['changed'] = stdout.find('Nothing to do.') == -1
```

Comment:
Did you test whether this works in check mode?

Reply:
I've missed the check-mode tests, I'll add this when I have some spare time.

Reply:
It did not work with check-mode. The new tests caught it and I've change this to return false when running in check-mode and otherwise read stdout.

Reply:
It might be better to return `changed=true` in then. `state=latest` can always update.

In any case, you should changed
```yaml
  check_mode:
    support: full
```
to
```yaml
  check_mode:
    support: partial
    details:
      - If O(state=latest), the module will always return C(changed=false).
```
(Resp. `C(changed=true)` if you change the behavior.)

Reply:
After some experimentation I've found a way to query `flatpak` if there are updates available for any of the named packages which is implemented in the new `_update_available()` function.

This should mean that check mode is fully supported now.

Reply:
I've reverted the `_update_available()` change as I wasn't able to figure out a consistent solution. I also changed the check mode default to `true` and updated the doc string as suggested.